### ISSUE TITLE
Cannot find module when explicit import extension

### ISSUE THREAD

**Issue by loynoir** [TypeStrong/ts-node#1494]
### Why explicit extension
Explicit import extension seems the ONLY convenient way to write dual module in typescript, supported by `tsc`.

### Expected Behavior
```txt
$ ts-node a.ts
42
```

### Actual Behavior
```txt
$ ts-node a.ts
Error: Cannot find module './b.js'
```

### Minimal reproduction
```ts
// a.ts
import b from './b.js'
console.log(b)
```

```ts
// b.ts
export default 42
```

### Specifications

* ts-node version:
* node version:
* TypeScript version:
* tsconfig.json, if you're using one:
```
{}
```
* Operating system and version:
* If Windows, are you using WSL or WSL2?:

**Comment by cspotcode** [comment]
Will be resolved by #1361

**Comment by cspotcode** [comment]
Closing as implemented already in recent releases of ts-node.  Depending on your project, you might need to enable `experimentalResolver: true`
